livekit-plugins/livekit-plugins-playai/livekit/plugins/playai/tts.py

```diff
@@ -0,0 +1,313 @@
+    APIConnectionError,
+    APIConnectOptions,
+    tokenize,
+    tts,
+    utils,
+)
+from pyht import AsyncClient as PlayHTAsyncClient  # type: ignore
+from pyht.client import Format, Language, TTSOptions  # type: ignore
+
+from .models import TTSModel
+
+
+@dataclass
+class _TTSOptions:
+    voice: str
+    format: Format
+    sample_rate: int
+    model: TTSModel | str
+    speed: float | None
+    language: Language
+    temperature: float | None
+    top_p: float | None
+    text_guidance: float | None
+    voice_guidance: float | None
+    style_guidance: float | None
+    repetition_penalty: float | None
+
+
+class TTS(tts.TTS):
+    def __init__(
+        self,
+        *,
+        api_key: str | None = None,
+        user_id: str | None = None,
+        voice: str = "s3://voice-cloning-zero-shot/775ae416-49bb-4fb6-bd45-740f205d20a1/jennifersaad/manifest.json",
+        language: str = "english",
+        sample_rate: int = 24000,
+        speed: float = 1.0,
+        model: TTSModel | str = "Play3.0-mini",
+        temperature: float | None = None,
+        top_p: float | None = None,
+        text_guidance: float | None = None,
+        voice_guidance: float | None = None,
+        style_guidance: float | None = None,
+        repetition_penalty: float | None = None,
+    ) -> None:
+        """
+        Initialize the PlayAI TTS engine.
+
+        Args:
+            api_key (str): The PlayAI API key. Can be set via environment variable PLAYAI_API_KEY.
+            user_id (str): The PlayAI user ID. Can be set via environment variable PLAYAI_USER_ID.
+            voice (str): A URL pointing to a Play voice manifest file. (e.g. "s3://voice-cloning-zero-shot/775ae416-49bb-4fb6-bd45-740f205d20a1/jennifersaad/manifest.json").
+            language (str): The language of the text. Default is 'english'.
+            sample_rate (int): The sample rate in Hz. Options are 8000, 16000, 24000, 44100, 48000.
+            speed (float): The speed of the audio. Default is 1.0.
+            model (str): The voice engine to use. Default is "Play3.0-mini".
+            > The following options are inference-time hyperparameters of the text-to-speech model; if unset, the model will use default values chosen by PlayAI.
+            temperature (float): The temperature of the model.
+            top_p (float): The top-p value of the model.
+            text_guidance (float): The text guidance of the model.
+            voice_guidance (float): The voice guidance of the model.
+            style_guidance (float): (Play3.0-mini-http and Play3.0-mini-ws only) The style guidance of the model.
+            repetition_penalty (float): The repetition penalty of the model.
+        """
+        super().__init__(
+            capabilities=tts.TTSCapabilities(
+                streaming=True,
+            ),
+            sample_rate=sample_rate,
+            num_channels=1,
+        )
+
+        api_key = api_key or os.environ.get("PLAYAI_API_KEY")
```

Comment:
See my other comment on `tests.yml`; you want to use play.ht credentials here, not play.ai

Reply:
I know this distinction is confusing and I'm sorry! In the future we hope to smooth all of this out.